eventmesh-metrics-plugin/eventmesh-metrics-api/src/main/java/org/apache/eventmesh/metrics/api/model/AbstractMetric.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.eventmesh.metrics.api.model;
+
+public abstract class AbstractMetric implements Metric {
+
+    private InstrumentFurther further;
```

Comment:
The classes under `org.apache.eventmesh.metrics.api.model` are too many and their inheritance relationship seems a little messy.

>This comment is a copy of https://github.com/apache/eventmesh/pull/3446#discussion_r1450994036

Reply:
Would it be better to extract implementation classes to a sub-package?

Reply:
There's no need to put this in a subclass; this covers all implementations. Using an abstract class makes it easier for future extensions.

Reply:
I meant to move files location without making changes to the class. The directory structure may like this:

```
model/
    counters/
        CounterImplA
        CounterImplB
    xxx/
        xxxImplA
        xxxImplB
CounterInterface
xxxInterface
xxxAbstractClassA
xxxAbstractClassB
```

It may be easier to find out how many interfaces and abstract classes are under this package and locate them.

Reply:
The `model` package is mainly used to store metrics. Personally, I don't think it's necessary to further classify the types of metrics. They can be uniformly placed in this package. Similarly, this package basically includes metrics that adhere to the existing OpenTelemetry specification. Being too granular makes the code look bloated.